server/src/main/java/org/opensearch/search/aggregations/metrics/CardinalityAggregatorFactory.java

```diff
@@ -53,19 +54,48 @@
  */
 class CardinalityAggregatorFactory extends ValuesSourceAggregatorFactory {
 
+    /**
+     * Execution mode for cardinality agg
+     *
+     * @opensearch.internal
+     */
+    public enum ExecutionMode {
+        DIRECT,
+        ORDINALS;
+
+        ExecutionMode() {}
+
+        public static ExecutionMode fromString(String value) {
+            try {
+                return ExecutionMode.valueOf(value.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Unknown `execution_hint`: [" + value + "], expected any of [direct, ordinals]");
```

Comment:
nit: Just to keep the formatting consistent with other places of `IllegalArgumentException` calls in code, how about:

```
throw new IllegalArgumentException("Unknown execution_hint [" + value + "], expected any of [direct, ordinals]");
```

(basically I was just getting rid of backticks enclosure)